### PR TITLE
Add `optimiser` test

### PR DESCRIPTION
**Pull request by Sjoerd1993**
Adds some tests to the `optimiser` module.

At the moment just does some minimal testing, where it checks the end-result given mock values for ```Optimiser.__optimise```. Some methods are not covered (still aiming for 100% here), and the ones that are need more tests to check edge cases as well for instance. 

Also needs clean up and some documentation (in the form of comments/docstrings) clearly. 

Closes issue #28

### CODE REVIEW

**Comment by Sjoerd1993** [comment]
This should be mostly ready now, it doesn't test the _optimise static method, but that's basically just a call to a different module. I was also slightly uncertain about the `func` methods that are tested here, the tests themselves for those functions are not super sophisticated, but just as a general test I decided to test against a known value from a previous run where I just outputted the values using the debugger.

File: hogben/tests/test_optimise.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    x = [0.3, 1.3, 0.8, 0.2]  # [angle, angle, time, time]
+    num_angles = 2
+    contrasts = [3, 14, -2]
+    points = 100
+    total_time = 10000
+
+    optimiser = Optimiser(refnx_sample)
+    result = optimiser._angle_times_func(x, num_angles, contrasts, points,
+                                         total_time)
+    expected_result = -1.7721879778537162
```
Comment (jfkcooper) [inline hogben/tests/test_optimise.py:102]: expected results have quite a large number of digits given you only want a `rtol` of `1e-6` but it really doesn't affect anything so leave it

File: hogben/tests/test_optimise.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    optimiser = Optimiser(BilayerDMPC())
+    result = optimiser._contrasts_func(x, num_contrasts, angle_splits,
+                                       total_time)
+    expected_result = -2.557881096447291
+    np.testing.assert_allclose(result, expected_result, rtol=1e-06)
+
+
+def test_underlayers_func():
+    """Checks that the _underlayers_func method gives the correct result"""
+    x = [50, 20, -5, 10]  # [thickness, thickness, SLD, SLD]
```
Comment (jfkcooper) [inline hogben/tests/test_optimise.py:122]: even in tests I'm not a fan of using `x` as a variable name

**Review by jfkcooper — CHANGES_REQUESTED** [review]
only one "real" comment, otherwise all good

File: hogben/tests/test_optimise.py (hunk 1 above)
Comment (Sjoerd1993) [reply hogben/tests/test_optimise.py:102]: Yeah, I increased the tolerance when I noticed that the test started failing when running in some different order, the issue there lies with the randomness due to the poisson "noise" in the data. So I upped the tolerance and the timing until I got a consistently passing test. The default value for `rtol` is `1e-7` by the way. Will do some testing to see if I can get this to a bit more reasonable values. 

Either way, I agree that so many significant digits make little sense when the tolerance is not in the area of this. So will also amend that a bit.

File: hogben/tests/test_optimise.py (hunk 2 above)
Comment (Sjoerd1993) [reply hogben/tests/test_optimise.py:122]: The `x` as variable name is simply what the tested function uses as well. But I agree that it's a poor choice of naming, which maybe is even apparant from the fact that I needed to add a clarifying comment after the assignments. 

Will ammend these names.

**Review by jfkcooper — APPROVED** [review]
